src/main/java/de/tum/in/www1/artemis/web/rest/settings/ide/IdeSettingsResource.java

```diff
@@ -0,0 +1,169 @@
+package de.tum.in.www1.artemis.web.rest.settings.ide;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.domain.settings.ide.Ide;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMapping;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.settings.IdeRepository;
+import de.tum.in.www1.artemis.repository.settings.UserIdeMappingRepository;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+import de.tum.in.www1.artemis.web.rest.dto.settings.ide.IdeDTO;
+import de.tum.in.www1.artemis.web.rest.dto.settings.ide.IdeMappingDTO;
+
+/**
+ * REST controller for managing Ide Settings (IdeSettings).
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class IdeSettingsResource {
+
+    private static final Logger log = LoggerFactory.getLogger(IdeSettingsResource.class);
+
+    private final UserIdeMappingRepository userIdeMappingRepository;
+
+    private final IdeRepository ideRepository;
+
+    private final UserRepository userRepository;
+
+    public IdeSettingsResource(UserIdeMappingRepository userIdeMappingRepository, IdeRepository ideRepository, UserRepository userRepository) {
+        this.userIdeMappingRepository = userIdeMappingRepository;
+        this.ideRepository = ideRepository;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * GET ide-settings/predefined: get the predefined IDEs
+     *
+     * @return the predefined ides as IdeDTO
+     */
+    @GetMapping("ide-settings/predefined")
+    @EnforceAtLeastStudent
+    public ResponseEntity<IdeDTO[]> getPredefinedIdes() {
+        log.debug("REST request to get predefined IDEs");
+
+        IdeDTO[] ideRecords = Arrays.stream(Ide.PREDEFINED_IDES).map(IdeDTO::new).toArray(IdeDTO[]::new);
+        log.debug("Successfully queried predefined IDEs");
+
+        return ResponseEntity.ok(ideRecords);
```

Comment:
I think you could use `ResponseEntity<List<IdeDTO>>` instead of an array here.

At least that's how it is done for other DTOs in Artemis, I haven't seen `ResponseEntity` with an array so far.

(For the client you nothing needs to be changed).
Any opinions on that?

Reply:
Yeah it seems lists are more commonly used, ill change that

Reply:
done